packages/@dcl/ecs/tools/protocol-buffer-generation/generateIndex.ts

```diff
@@ -11,6 +13,9 @@ function importComponent(component: Component) {
 }
 
 function defineComponent(component: Component) {
+  if (component.componentId === TransformComponent.componentId) {
```

Comment:
what if instead of doing this wrapper, we create a new param to `defineComponent` with defaultValues.
We can use this for custom components also

```ts
const PlayerType = MapType({
  velocity: Vector3,
  position: Vector3,
  name: EcsString,
  etc: MapType({ ... })
})

const defaultValues = {
  velocity: { x: 0, y: 0, z: 0 },
  position: { x: 0, y: 0, z: 0 },
}

const PlayerComponent = engine.defineComponent(PlayerType, 1000, defaultValues)
PlayerComponent.create(entity, { name: 'myName' })
```

Reply:
It may kill the performance?

Reply:
https://github.com/decentraland/js-sdk-toolchain/pull/230

Reply:
This is a WIP approach; it extends the functionality to all the Schemas and may be helpful. Schemas always serialize their fields so that the performance won't be affected.